src/crypto/crypto_keygen.h

```diff
@@ -174,7 +169,10 @@ struct KeyPairGenTraits final {
     if (!EVP_PKEY_keygen(ctx.get(), &pkey))
       return KeyGenJobStatus::FAILED;
 
-    params->key = ManagedEVPPKey(EVPKeyPointer(pkey));
+    auto data = KeyObjectData::CreateAsymmetric(KeyType::kKeyTypePrivate,
+                                                EVPKeyPointer(pkey));
+    if (!data) return KeyGenJobStatus::FAILED;
```

Comment:
How come we didn't need this if statement previously?

```suggestion
    if (UNLIKELY(!data)) return KeyGenJobStatus::FAILED;
```

Reply:
We likely did, it just wasn't there.